frontend/src/NavBar.js

```diff
@@ -1,59 +1,42 @@
-import React, { Component } from 'react'
+import React, { useEffect, useState } from 'react'
 import { Link } from 'react-router-dom'
 import SearchBar from './SearchBar'
 
 /**
  * The navigation bar at the top of the page, containing the logo, search bar, and cart icon.
  */
-class NavBar extends Component {
-  constructor(props) {
-    super(props)
 
-    this.state = {
-      courseCount: Object.keys(localStorage).filter(a => !a.startsWith('meta-'))
-        .length,
+export const NavBar = () => {
+  const getCourseCount = () =>
+    Object.keys(localStorage).filter(a => !a.startsWith('meta-')).length
+  const [courseCount, setCourseCount] = useState(getCourseCount())
+  const onStorageChange = () => setCourseCount(getCourseCount())
+
+  useEffect(() => {
+    window.addEventListener('storage', onStorageChange)
+    window.onCartUpdated = onStorageChange
+    return () => {
+      window.removeEventListener('storage', onStorageChange)
+      window.onCartUpdated = null
     }
-
-    this.onStorageChange = this.onStorageChange.bind(this)
-  }
-
-  onStorageChange() {
-    this.setState({
-      courseCount: Object.keys(localStorage).filter(a => !a.startsWith('meta-'))
-        .length,
-    })
-  }
-
-  componentDidMount() {
-    window.addEventListener('storage', this.onStorageChange)
-    window.onCartUpdated = this.onStorageChange
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('storage', this.onStorageChange)
-    window.onCartUpdated = null
-  }
-
-  render() {
-    return (
-      <div id="header">
-        <span className="float-left">
-          <Link to="/" title="Go to Penn Course Review Home">
-            <div id="logo" />
-          </Link>
-          <SearchBar />
-        </span>
-        <span className="float-right">
-          <Link to="/cart" id="cart-icon" title="Course Cart">
-            <i id="cart" className="fa fa-shopping-cart" />
-            {this.state.courseCount > 0 && (
-              <span id="cart-count">{this.state.courseCount}</span>
-            )}
-          </Link>
-        </span>
-      </div>
-    )
-  }
+  }, [])
+
+  return (
+    <div id="header">
+      <span className="float-left">
+        <Link to="/" title="Go to Penn Course Review Home">
+          <div id="logo" />
+        </Link>
+        <SearchBar />
+      </span>
+      <span className="float-right">
+        <Link to="/cart" id="cart-icon" title="Course Cart">
```

Comment:
Also would be good to make routes stored in a shared constants file, that can also be a different PR